### ISSUE TITLE
BigQuery Keyword Rules L010 and L040

### ISSUE THREAD

**Issue by rileyrunnoe** [sqlfluff/sqlfluff#2912]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When setting a capitalization policy for rule L040, if that rule interferes with the expectations for rule L010, the linter will either say one rule is wrong or the other.

### Expected Behaviour

I believe that rule L040 should supersede L010, namely that if you have a setting for bools and nulls, it shouldn't be overridden by L010. However, that is probably a decision for a maintainer. I am more so looking for consistency to not have these rules interfere with each other. 

### Observed Behaviour

Please see the how to reproduce section.

### How to reproduce

I am running sqlfluff 0.11.1 with the following .sqlfluff file

I will attempt to lint the following SQL:

```sql
SELECT TRUE FROM {{ this }} AS RET WHERE RET.SOURCE_FILE_NAME = RAW.SOURCE_FILE_NAME
```
But it will say that it is violating rule L040. When i go to fix the error, the resulting SQL looks as follows:

```sql
SELECT true FROM {{ this }} AS RET WHERE RET.SOURCE_FILE_NAME = RAW.SOURCE_FILE_NAME
```
But that now violates rule L010. 

### Dialect

BigQuery

### Version

v 0.11.1

### Configuration

[sqlfluff]
#dialect
dialect = bigquery
templater = jinja

#rule ignoring and exclusion
#Allow explicit ELSE NULL statements
#Allow table aliases
exclude_rules = L035,L031

# NB: This config will only apply in the root folder.
sql_file_exts = .sql

[sqlfluff:rules]
comma_style = leading
indent_unit = space
max_line_length = 120
tab_space_size = 4

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = lower

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = True

[sqlfluff:templater:jinja]
load_macros_from_path = macros

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rileyrunnoe** [comment]
Willing to work on this, but I am not really sure where to start. I think it might be a keyword issue, but could use some guidance on where to look.

**Comment by tunetheweb** [comment]
The problem is that `TRUE`, `FALSE` and `NULL` are all Keywords:

```py
    NullLiteralSegment=StringParser(
        "null", KeywordSegment, name="null_literal", type="literal"
    ),
    TrueSegment=StringParser(
        "true", KeywordSegment, name="boolean_literal", type="literal"
    ),
    FalseSegment=StringParser(
        "false", KeywordSegment, name="boolean_literal", type="literal"
    ),
```

Therefore they fall under L010 (Keywords) and L040 (Boolean Identifiers), and if you have these rules set up in a opposing manner (whether directly, or by using defaults) then they clash.

I think the solution is to exclude these from L010,  as they have their own rule.

You can see L040 looks at these types:

https://github.com/sqlfluff/sqlfluff/blob/c915ddf418c20c3a752509f4c0dd4eb96b591bfd/src/sqlfluff/rules/L040.py#L56-L59

so all you need is add something like this to rule L010 at line 68:

```py
        # Skip boolean and null literals as they have their own rule (L040)
        if context.segment.name in ["null_literal", "boolean_literal"]:
            return LintResult(memory=context.memory)
```

Then add a test case or two to L010.yml and good to submit as a PR!

**Comment by rileyrunnoe** [comment]
Thank you all!
